packages/bvaughn-architecture-demo/components/console/Input.tsx

```diff
@@ -25,10 +29,18 @@ export default function Input() {
     searchStateVisibleRef.current = searchState.visible;
   }, [searchState.visible]);
 
-  const currentPause = useCurrentPause();
   const { selectedPauseAndFrameId } = useContext(SelectedFrameContext);
-  const frameId = selectedPauseAndFrameId?.frameId ?? currentPause?.stack?.[0] ?? null;
-  const pauseId = selectedPauseAndFrameId?.pauseId ?? currentPause?.pauseId ?? null;
+  let pauseId: PauseId | null = useCurrentPauseIdSuspense();
+  let frameId: FrameId | null = null;
+  if (selectedPauseAndFrameId) {
+    pauseId = selectedPauseAndFrameId.pauseId;
+    frameId = selectedPauseAndFrameId.frameId;
+  } else {
+    if (pauseId) {
+      const frames = getFramesSuspense(replayClient, pauseId);
+      frameId = frames?.[0]?.frameId ?? null;
+    }
```

Comment:
Seems like we would ideally _not_ suspend here until after we'd confirmed the pause and frame ids didn't match.
```suggestion
  let pauseId: PauseId | null = null;
  let frameId: FrameId | null = null;
  if (selectedPauseAndFrameId) {
    pauseId = selectedPauseAndFrameId.pauseId;
    frameId = selectedPauseAndFrameId.frameId;
  } else {
    pauseId = useCurrentPauseIdSuspense();
    if (pauseId) {
      const frames = getFramesSuspense(replayClient, pauseId);
      frameId = frames?.[0]?.frameId ?? null;
    }
```

Reply:
`useCurrentPauseIdSuspense()` is a hook and would be called conditionally with this suggestion. So I removed it and do the hook part unconditionally and the suspending part conditionally now.